### PR TITLE
feat: update xSync's UI/UX

### PR DESCRIPTION
**Pull request by Runjuu**

### CODE REVIEW

**Comment by Songkeys** [comment]
Opening without wallet connected, it keeps spinning: https://crossbell-io-git-feat-sync-ui-crossbell.vercel.app/sync

**Comment by Runjuu** [comment]
> Opening without wallet connected, it keeps spinning: https://crossbell-io-git-feat-sync-ui-crossbell.vercel.app/sync

@Songkeys Updated ;)

**Comment by Songkeys** [comment]
When this model opened, I can't click the outside overlay to close it.

image

**Comment by Runjuu** [comment]
> When this model opened, I can't click the outside overlay to close it.
> 
> image

@Songkeys Updated. This modal can now be closed by clicking outside.

File: components/pages/sync/platforms-section/welcome.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+			style={style}
+			className="p-24px rounded-24px border border-1 border-[#E1E8F7] mb-24px"
+		>
+			<h4 className="mt-0 mb-16px text-[#262626] font-400 text-22px leading-28px">
+				Welcome!!!
+			</h4>
+			<p className="m-0 font-400 text-16px leading-24px text-[#262626] children:text-[#5B89F7]">
+				Try to bind a platform to sync your content! Visiting our{" "}
+				<a
+					href="https://sync-landing-page-i51q.vercel.app"
```
Comment (Songkeys) [inline components/pages/sync/platforms-section/welcome.tsx:22]: Is this the final version?

**Comment by Songkeys** [comment]
Loading hangs when clicking the switch

image

**Comment by Songkeys** [comment]
image

Shouldn't be called "Capacity" (meaning the upper limit). Something like "Used Storage" is more suitable.

**Comment by Songkeys** [comment]
image


image

I think here needs a DisplayName.

**Comment by Songkeys** [comment]
image

account => Account

**Comment by Songkeys** [comment]
image

there should be a "cursor-point" for the switch

**Comment by Songkeys** [comment]
Some wordings need to change.

image

**Comment by Songkeys** [comment]
image

image

I do think we could just say "Please unbind each platform before turning off sync." in a single step. If users hasn't done yet, just don't let them continue.

**Comment by Songkeys** [comment]
image

What's the meaning of "Go to medium" as a primary button? I think a single "OK" button to close the modal and let user stay and keep browsing would be more meaningful.

**Comment by Songkeys** [comment]
image

platform => platforms

**Comment by AtlasQuan** [comment]
> image
> 
> image
> 
> I do think we could just say "Please unbind each platform before turning off sync." in a single step. If users hasn't done yet, just don't let them continue.

``Operator`` is the one the people granting permission to do operations on his own behalf. That is simply a record on-chain and fully controlled by the user. But ``binding platforms to sync`` is permission-based on a centralized server. Then it is "theoretically" not fully controlled by the user. 

From this point, I believe it's not a good practice on UI/UX to stop users to continue turning off sync, which is an operation he has full control to do, because of "unbinding platforms", which is an operation he doesn't have full control to do.

File: components/pages/sync/platforms-section/welcome.tsx (hunk 1 above)
Comment (Runjuu) [reply components/pages/sync/platforms-section/welcome.tsx]: Replaced with `https://xsync.crossbell.io/`

**Comment by AtlasQuan** [comment]
> @AtlasQuan you reason behind the free-control topic. I reason behind the **faultless** and **simple** UX. If it won't cause any "Serious Issues", I'm okay with it. Otherwise, why add more mental burden to users on this step? If user want free control, they can interact on the scan directly without worrying about any further "Serious Issues".

But isn't telling people to interact on the scan a mental burden? UI is not only made to experience **service** in a comfortable way but also made to give users control of the data they **own**.  But if we could have both, that definitely is better. So honestly I prefer the solution is: make this issue not "serious", which means the "sync server" is required to gracefully handle the "non-authorized operator". cc @Candinya 

So how about keeping the UI as it is until the sync server gracefully handles the non-authorized operator?  Then we change the UX to only have one alert box to decide "cancel" or "I've decided"?

**Comment by Songkeys** [comment]
> But isn't telling people to interact on the scan a mental burden?

Provided that "user want *fully* free control". You have to either implement all controllers on the UI or invite them to use the scan / interact with the contract directly. Sounds not good? So why bother implementing such "*partial* free control" at the current early stage?

> UI is not only made to experience service in a comfortable way

Do agree.

> but also made to give users control of the data they own

But the current implementation has a flaw to the system. You provide the way but warn user not to do it. It's a broken design and experience. And, it doesn't actually mean anything to users. I doubt there is any user actually caring about this level of free control. But there must be someone care about flawless UX.

> But if we could have both, that definitely is better.

You can't have your cake and eat it too. Especially on an early stage. I think UX is more important and should be focused now. *Partial* free control with a flaw implementation is on the opposite.

> So honestly I prefer the solution is: make this issue not "serious", which means the "sync server" is required to gracefully handle the "non-authorized operator". cc @Candinya

I don't know how hard it is to fix. But honestly I don't think this topic deserves further development. There's more to be focused.

> So how about keeping the UI as it is until the sync server gracefully handles the non-authorized operator? Then we change the UX to only have one alert box to decide "cancel" or "I've decided"?

Yeah. Though I feel not quite right about this modal, I'd like to see everything kept as-is because this topic doesn't deserve this verbose discussion and implementation.

Anyway, I was wanting to express that I love more focus on the important UX but not other subtle concepts introduced to break it. The above discussion is a breakdown of the reasoning and debate.

Let's give a conclusion: this PR is actually good enough to merge. What do you think?

**Comment by Songkeys** [comment]
@Runjuu One more modification might needs to be done.

image

**Comment by AtlasQuan** [comment]
> Provided that "user want _fully_ free control". You have to either implement all controllers on the UI or invite them to use the scan / interact with the contract directly. Sounds not good? So why bother implementing such "_partial_ free control" at the current early stage?

That's because "disable the operator" is a serious control -- "operator" has full access to post notes on behalf of the user. That's different from other controls, e.g. transferring his character to another address. Especially here's already the context to manage operator.

> But the current implementation has a flaw to the system. You provide the way but warn user not to do it. It's a broken design and experience. And, it doesn't actually mean anything to users. I doubt there is any user actually caring about this level of free control. But there must be someone care about flawless UX.

I feel the reason behind that "inconsistency" is that, from my philosophy of DApp, the only trusted source is the contract, whereas other service provider(sync service) is not. And dapp should not stand with the service provider but with the user. Though in our case, "we" are both the dapp builder and the service provider, as a dapp builder still it matters to always leave an alternative for users to interact with the contract without relying on the trust of servers.

For a traditional model, the app builder stands with the service provider, so that UX is always consistent in reflecting what the service providers ask the user to do.  But Web3 dapps are supposed to have a higher priority to listen from the user, than obeying the service provider's intention.

In our case, it doesn't wanna the user to "turn sync off" before unbinding all platforms; however, as the dapp builder, we wanna do what the user decides to do.

(So... basically, to some degree my philosophy of DApps is against what is usually said by people: "user doesn't know what he wants to do".)

> But honestly I don't think this topic deserves further development. There's more to be focused.

Sure... Cannot agree more.


> Anyway, I was wanting to express that I love more focus on the important UX but not other subtle concepts introduced to break it. The above discussion is a breakdown of the reasoning and debate.

Yeah! Great discussion and I could totally relate to your reasons behind it.

> Let's give a conclusion: this PR is actually good enough to merge. What do you think?

Absolutely.

**Comment by Runjuu** [comment]
> @Runjuu One more modification might need to be done. I'm not sure if the UI is ready tho.
> 
> image

@Songkeys Updated~

**Pull request merged**